bin/metrics-libcurl.rb

```diff
@@ -0,0 +1,169 @@
+#!/usr/bin/env ruby
+# frozen_string_literal: false
+
+#
+#   metrics-libcurl
+#
+# DESCRIPTION:
+#   Simple wrapper around libcurl for getting timing stats from the various phases
+#   of connecting to an HTTP/HTTPS server.
+#
+# OUTPUT:
+#   metric data
+#
+# PLATFORMS:
+#   Linux
+#
+# DEPENDENCIES:
+#   gem: sensu-plugin
+#   gem: typhoeus
+#
+# USAGE:
+#   #YELLOW
+#
+# NOTES:
+#   Based on: metrics-curl.rb
+#   by Joe Miller.
+#
+# LICENSE:
+#   Copyright 2019 Jef Spaleta
+#   Released under the same terms as Sensu (the MIT license); see LICENSE
+#   for details.
+#
+
+require 'socket'
+require 'English'
+require 'sensu-plugin/metric/cli'
+require 'typhoeus'
+require 'json'
+
+
+
+#
+# Libcurl Metrics
+#
+class LibcurlMetrics < Sensu::Plugin::Metric::CLI::Graphite
+  option :url,
+         short: '-u URL',
+         long: '--url URL',
+         description: 'valid cUrl url to connect (default: http://127.0.0.1:80/)',
+         default: 'http://127.0.0.1:80/'
+
+  option :scheme,
+         description: 'Metric naming scheme, text to prepend to metric',
+         short: '-s SCHEME',
+         long: '--scheme SCHEME',
+         default: "#{Socket.gethostname}.curl_timings"
+  option :debug,
+         description: 'Include debug output, should not use in production.',
+         short: '-d',
+         long: '--debug',
+         default: false
+  option :libcurl_options,
+         description: 'Libcurl Options as a key/value JSON string',
+         short: '-o JSON',
+         long: '--options  JSON',
+         default: '{}'
+  option :http_headers,
+         description: 'HTTP Request Headers as key/value JSON string',
+         short: '-H JSON',
+         long: '--headers  JSON',
+         default: '{}'
+  option :http_params,
+         description: 'HTTP Request Parameters as key/value JSON string',
+         short: '-P JSON',
+         long: '--params  JSON',
+         default: '{}'
+  option :http_response_error,
+         description: 'return critical status (2) if http response error status encountered (>= 400)',
+         short: '-c',
+         long: '--critical_http_error',
+         default: false
+  option :http_redirect_warning,
+         description: 'return warning status (1) if http response redirect status encountered (3xx)',
+         short: '-w',
+         long: '--warn_redirect',
+         default: false
+  option :help,
+         short: '-h',
+         long: '--help',
+         description: 'Show this message',
+         on: :tail,
+         boolean: true,
+         show_options: true
+
+  def usage_details
+    <<~USAGE
```

Comment:
looks like rubocop is falling back to trying with the 2.1 parser, lets add `TargetRubyVersion` to our rubocop to appease that.